src/lib/y2ntp_client/dialog/add_pool.rb

```diff
@@ -0,0 +1,91 @@
+require "yast"
+
+require "cwm/popup"
+
+Yast.import "Label"
+Yast.import "NtpClient"
+Yast.import "Stage"
+Yast.import "Popup"
+
+# Work around YARD inability to link across repos/gems:
+
+# @!macro [new] seeAbstractWidget
+#   @see http://www.rubydoc.info/github/yast/yast-yast2/CWM/AbstractWidget:${0}
+# @!macro [new] seeCustomWidget
+#   @see http://www.rubydoc.info/github/yast/yast-yast2/CWM/CustomWidget:${0}
+# @!macro [new] seeItemsSelection
+#   @see http://www.rubydoc.info/github/yast/yast-yast2/CWM/ItemsSelection:${0}
+# @!macro [new] seeDialog
+#   @see http://www.rubydoc.info/github/yast/yast-yast2/CWM/Dialog:${0}
+# @!macro [new] seePopup
+#   @see http://www.rubydoc.info/github/yast/yast-yast2/CWM/Popup:${0}
+
+module Y2NtpClient
+  module Dialog
+    # Dialog to add/edit ntp pool server
+    class AddPool < ::CWM::Popup
+      # Constructor
+      #
+      # @param address_widget [CWM::InputField]
+      # @param pool_type [Symbol]
+      def initialize(address_widget, pool_type)
+        textdomain "ntp-client"
+        @address_widget = address_widget
+        @address = @address_widget.value
+        @pool_chooser = pool_for(pool_type)
+      end
+
+      # @macro seeDialog
+      def title
+        _("Local ntp servers discovered")
+      end
+
+      # @macro seeDialog
+      def contents
+        VBox(
+          @pool_chooser
+        )
+      end
+
+      def next_handler
+        return :cancel if @pool_chooser.value.to_s.empty?
+        @address = @pool_chooser.value
+
+        :next
+      end
+
+      # @macro seeDialog
+      def next_button
+        ok_button_label
+      end
+
+      # @macro seeDialog
+      def run
+        result = super
+        @address_widget.value = @address if result == :next
+
+        result
+      end
+
+    private
+
+      # @macro seeDialog
+      def ok_button
+        PushButton(Id(:next), Opt(:default), ok_button_label)
+      end
+
+      def pool_for(type)
+        { local:  Widgets::LocalList,
```

Comment:
I don't know if this likes to Rubocop, but I prefer

```ruby
{
  local: ...,
  public: ...
}.fetch(...)
```

or even more readable, 

```ruby
def available_pools
 { local: ..., public: ... }
end

def pool_for(type)
  pool = available_pools.fetch(type) { Widgets::LocalList }
  pool.new(@address_widget.value)
end

Reply:
or even

```ruby
def available_pools
  @available_pools ||=
    begin
      pools = Hash.new(Widgets::Locallist)
      pools[:local] = ...
      pools[:public] = ...
      pools
    end
end 

def pools_for(type)
  pool = available_pools[type]
  pool.new(...)
end